src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -180,7 +180,9 @@ public void deserializeQueryContextJson(String data) {
       if (entriesNode != null && entriesNode.isArray()) {
           for (JsonNode entryNode : entriesNode) {
             QueryContextElement entry = deserializeQueryContextElement(entryNode);
-            merge(entry.id, entry.readTimestamp, entry.priority, entry.context);
+            if(entry != null){
+              merge(entry.id, entry.readTimestamp, entry.priority, entry.context);
+            }
```

Comment:
In the else part can we clear the cache and terminate the loop?

Reply:
Yeah. I forgot to push the commits on early termination and cache clear. Just pushed it.